File: java/web_app/handrolled/src/main/java/Presenters/TTTOptionPresenter.java
package Presenters;

public class TTTOptionPresenter {
    public static String generateOptions() {
        String htmlOptions = "<form class='start-button' action='/game' method=\"post\">";
        htmlOptions += "<small> player one </small>";
        htmlOptions += "<select name=\"playerOne\" size=\"1\">";
        htmlOptions += "<option value=" + "human> human" + "</option>";
        htmlOptions += "<option value=" + "computer> computer" + "</option></select>";
        htmlOptions += "<small> player two </small>";
        htmlOptions += "<select name=\"playerTwo\" size=\"1\">";
        htmlOptions += "<option value=" + "human> human" + "</option>";
        htmlOptions += "<option value=" + "computer> computer" + "</option></select>";
        htmlOptions += "<small> board size </small>";
        htmlOptions += "<select name=\"boardSize\" size=\"1\">";
        htmlOptions += "<option value=" + "3> 3" + "</option>";
        htmlOptions += "<option value=" + "4> 4" + "</option></br>";
        htmlOptions += "<div><input id=\"start_game\" type=\"submit\" value=\"Start Game\"></div>";
        htmlOptions += "</form>";
        return htmlOptions;
    }
}

File: ruby/web_app/sinatra/lib/presenters/ttt_option_presenter.rb
class TTTOptionPresenter
  def self.generate_options(options)
    html_options = '<form class="start-button" action="/game" method="post">'

    options.each do |key, value|
      html_options += "<small> #{key} </small>"
      html_options += "<select name=#{key} size='1'>"
      value.each do |option|
        html_options += "<option value =#{option}>#{option}</option>"
      end
      html_options += "</select>"
    end

    html_options += "<div><input id='start_game' type='submit' value='Start Game'></div>"
    html_options += '</form>'
  end
end
